Index: target/linux/ramips/image/Makefile
===================================================================
--- target/linux/ramips/image/Makefile (revision 38000)
+++ target/linux/ramips/image/Makefile (working copy)
@@ -291,6 +291,8 @@

 Image/Build/Profile/AIR3GII=$(call BuildFirmware/Default4M/$(1),$(1),air3gii,AIR3GII)

+Image/Build/Profile/AYYAR=$(call BuildFirmware/Default8M/$(1),$(1),ayyar,AYYAR,$(ralink_default_fw_size_8M))
+
 define BuildFirmware/UIMAGE_8M
	$(call MkCombineduImage,$(1),$(2),$(call mkcmdline,$(3),$(4),$(5)) $(call mkmtd/$(6),$(mtdlayout_8M)),$(kernel_size_8M),$(rootfs_size_8M),$(7))
 endef
@@ -498,6 +498,7 @@
	$(call Image/Build/Profile/ALL5002,$(1))
	$(call Image/Build/Profile/ALL5003,$(1))
	$(call Image/Build/Profile/ARGUS_ATP52B,$(1))
+	$(call Image/Build/Profile/AYYAR,$(1))
	$(call Image/Build/Profile/ASL26555,$(1))
	$(call Image/Build/Profile/AWM002-EVB,$(1))
	$(call Image/Build/Profile/BC2,$(1))
Index: target/linux/ramips/rt305x/profiles/ayyar.mk
===================================================================
--- target/linux/ramips/rt305x/profiles/ayyar.mk	(revision 0)
+++ target/linux/ramips/rt305x/profiles/ayyar.mk	(working copy)
@@ -0,0 +1,18 @@
+#
+# Copyright (C) 2013 Ayyari Abdennaby <ayyari@me.com>
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+define Profile/AYYAR
+	NAME:=AYYAR Realtime Web Control
+	PACKAGES:=kmod-usb-core kmod-usb-ohci kmod-usb2 kmod-ledtrig-usbdev \
+	kmod-i2c-core kmod-i2c-ralink kmod-i2c-gpio i2c_tool vsftpd
+endef
+
+define Profile/AYYAR/Description
+	Package set for  AYYAR Real time web control board
+endef
+
+$(eval $(call Profile,AYYAR))
Index: target/linux/ramips/base-files/lib/ramips.sh
===================================================================
--- target/linux/ramips/base-files/lib/ramips.sh	(revision 38000)
+++ target/linux/ramips/base-files/lib/ramips.sh	(working copy)
@@ -25,6 +25,9 @@
	*"AirLive Air3GII")
		name="air3gii"
		;;
+	*"AYYAR")
+ 		name="ayyar"
+ 		;;
	*"Edimax BR-6425")
		name="br6425"
		;;
Index: target/linux/ramips/base-files/lib/upgrade/platform.sh
===================================================================
--- target/linux/ramips/base-files/lib/upgrade/platform.sh	(revision 38000)
+++ target/linux/ramips/base-files/lib/upgrade/platform.sh	(working copy)
@@ -25,6 +25,7 @@
	all5003 | \
	asl26555 | \
	awm002-evb | \
+	ayyar | \
	bc2 | \
	broadway | \
	carambola | \
Index: target/linux/ramips/base-files/lib/preinit/06_set_iface_mac
===================================================================
--- target/linux/ramips/base-files/lib/preinit/06_set_iface_mac	(revision 38000)
+++ target/linux/ramips/base-files/lib/preinit/06_set_iface_mac	(working copy)
@@ -58,6 +58,7 @@
	all5002 |\
	all5003 |\
	awm002-evb |\
+	ayyar |\
	carambola |\
	dir-615-h1 |\
	fonera20n |\
Index: target/linux/ramips/base-files/etc/diag.sh
===================================================================
--- target/linux/ramips/base-files/etc/diag.sh	(revision 38000)
+++ target/linux/ramips/base-files/etc/diag.sh	(working copy)
@@ -18,6 +18,9 @@
	asl26555)
		status_led="asl26555:green:power"
		;;
+	ayyar)
+		status_led="ayyar:green:power"
+		;;
	br6524n)
		status_led="edimax:blue:power"
		;;
Index: target/linux/ramips/base-files/etc/hotplug.d/firmware/10-rt2x00-eeprom
===================================================================
--- target/linux/ramips/base-files/etc/hotplug.d/firmware/10-rt2x00-eeprom	(revision 38000)
+++ target/linux/ramips/base-files/etc/hotplug.d/firmware/10-rt2x00-eeprom	(working copy)
@@ -62,6 +62,7 @@
	all5003 | \
	argus-atp52b | \
	awm002-evb | \
+	ayyar | \
	bc2 | \
	br6425 | \
	broadway | \
Index: target/linux/ramips/base-files/etc/uci-defaults/01_leds
===================================================================
--- target/linux/ramips/base-files/etc/uci-defaults/01_leds	(revision 38000)
+++ target/linux/ramips/base-files/etc/uci-defaults/01_leds	(working copy)
@@ -41,6 +41,9 @@
		set_wifi_led "asl26555:green:wlan"
		set_usb_led "asl26555:green:3g"
		;;
+	ayyar)
+		set_wifi_led "rt2800pci-phy0::radio"
+		;;
	bc2)
		set_usb_led "bc2:blue:usb"
		;;
Index: target/linux/ramips/base-files/etc/uci-defaults/02_network
===================================================================
--- target/linux/ramips/base-files/etc/uci-defaults/02_network	(revision 38000)
+++ target/linux/ramips/base-files/etc/uci-defaults/02_network	(working copy)
@@ -42,6 +42,7 @@
	all5002 | \
	all5003 | \
	awm002-evb | \
+	ayyar | \
	broadway | \
	wnce2001)
		ucidef_add_switch "switch0" "1" "0"
