Index: target/linux/ramips/image/Makefile
===================================================================
--- target/linux/ramips/image/Makefile (revision 36362)
+++ target/linux/ramips/image/Makefile (working copy)
@@ -269,6 +269,8 @@
 argus_atp52b_mtd_size=7995392
 Image/Build/Profile/ARGUS_ATP52B=$(call BuildFirmware/CustomFlash/$(1),$(1),argus_atp52b,ARGUS_ATP52B,$(argus_atp52b_mtd_size))
 
+Image/Build/Profile/AYYAR=$(call BuildFirmware/Default8M/$(1),$(1),ayyar,AYYAR,$(ralink_default_fw_size_8M))
+
 Image/Build/Profile/BC2=$(call BuildFirmware/Default8M/$(1),$(1),bc2,BC2)

 broadway_mtd_size=7929856
@@ -424,6 +424,7 @@
 $(call Image/Build/Profile/ALL0256N,$(1))
	$(call Image/Build/Profile/ALL5002,$(1))
	$(call Image/Build/Profile/ARGUS_ATP52B,$(1))
+	$(call Image/Build/Profile/AYYAR,$(1))
	$(call Image/Build/Profile/BC2,$(1))
	$(call Image/Build/Profile/BROADWAY,$(1))
	$(call Image/Build/Profile/CARAMBOLA,$(1))
Index: target/linux/ramips/rt305x/profiles/ayyar.mk
===================================================================
--- target/linux/ramips/rt305x/profiles/ayyar.mk	(revision 0)
+++ target/linux/ramips/rt305x/profiles/ayyar.mk	(working copy)
@@ -0,0 +1,18 @@
+#
+# Copyright (C) 2013 Ayyari Abdennaby <ayyari@me.com>
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+define Profile/AYYAR
+	NAME:=AYYAR Realtime Web Control
+	PACKAGES:=kmod-usb-core kmod-usb-ohci kmod-usb2 kmod-ledtrig-usbdev \
+	kmod-i2c-core kmod-i2c-gpio kmod-gpio-pcf857x
+endef
+
+define Profile/AYYAR/Description
+	Package set for  AYYAR Real time web control board
+endef
+
+$(eval $(call Profile,AYYAR))
Index: target/linux/ramips/base-files/lib/ramips.sh
===================================================================
--- target/linux/ramips/base-files/lib/ramips.sh	(revision 36362)
+++ target/linux/ramips/base-files/lib/ramips.sh	(working copy)
@@ -39,6 +39,9 @@
	*"AirLive Air3GII")
		name="air3gii"
		;;
+	*"AYYAR")
+ 		name="ayyar"
+ 		;;
	*"Edimax BR-6425")
		name="br6425"
		;;
Index: target/linux/ramips/base-files/lib/upgrade/platform.sh
===================================================================
--- target/linux/ramips/base-files/lib/upgrade/platform.sh	(revision 36362)
+++ target/linux/ramips/base-files/lib/upgrade/platform.sh	(working copy)
@@ -22,6 +22,7 @@
	all0239-3g | \
	all0256n | \
	all5002 | \
+	ayyar | \
	bc2 | \
	broadway | \
	carambola | \
Index: target/linux/ramips/base-files/lib/preinit/06_set_iface_mac
===================================================================
--- target/linux/ramips/base-files/lib/preinit/06_set_iface_mac	(revision 36362)
+++ target/linux/ramips/base-files/lib/preinit/06_set_iface_mac	(working copy)
@@ -51,6 +51,7 @@
	all0239-3g |\
	all0256n |\
	all5002 |\
+	ayyar | \
	carambola |\
	dir-615-h1 |\
	fonera20n |\
Index: target/linux/ramips/base-files/etc/diag.sh
===================================================================
--- target/linux/ramips/base-files/etc/diag.sh	(revision 36362)
+++ target/linux/ramips/base-files/etc/diag.sh	(working copy)
@@ -15,6 +15,9 @@
	argus-atp52b)
		status_led="argus-atp52b:green:run"
		;;
+	ayyar)
+		status_led="ayyar:green:power"
+		;;
	br6524n)
		status_led="edimax:blue:power"
		;;
Index: target/linux/ramips/base-files/etc/hotplug.d/firmware/10-rt2x00-eeprom
===================================================================
--- target/linux/ramips/base-files/etc/hotplug.d/firmware/10-rt2x00-eeprom	(revision 36362)
+++ target/linux/ramips/base-files/etc/hotplug.d/firmware/10-rt2x00-eeprom	(working copy)
@@ -59,6 +59,7 @@
	all0256n | \
	all5002 | \
	argus-atp52b | \
+	ayyar | \
	bc2 | \
	br6425 | \
	broadway | \
Index: target/linux/ramips/base-files/etc/uci-defaults/01_leds
===================================================================
--- target/linux/ramips/base-files/etc/uci-defaults/01_leds	(revision 36362)
+++ target/linux/ramips/base-files/etc/uci-defaults/01_leds	(working copy)
@@ -35,6 +35,9 @@
		ucidef_set_led_rssi "rssihigh" "RSSIHIGH" "all0256n:green:rssihigh" "wlan0" "70" "100" "-69" "8"
		set_wifi_led "rt2800pci-phy0::radio"
		;;
+	ayyar)
+		set_wifi_led "rt2800pci-phy0::radio"
+		;;
	bc2)
		set_usb_led "bc2:blue:usb"
		;;
Index: target/linux/ramips/base-files/etc/uci-defaults/02_network
===================================================================
--- target/linux/ramips/base-files/etc/uci-defaults/02_network	(revision 36362)
+++ target/linux/ramips/base-files/etc/uci-defaults/02_network	(working copy)
@@ -41,6 +41,7 @@
	w150m | \
	all0256n | \
	all5002 | \
+	ayyar | \
	broadway | \
	wnce2001)
		ucidef_add_switch "switch0" "1" "0"
